otter/test/worker/test_launch_server_v1.py

```diff
@@ -912,9 +912,14 @@ def setUp(self):
             'otter.worker.launch_server_v1.generate_server_name')
         self.generate_server_name.return_value = 'as000000'
 
+        # reset it to empty in beginning of test as it is module-wise
+        # info retained in every test
+        launch_server_v1._semaphores = {}
```

Comment:
This looks dodgy to me. Consider using [`patch`](https://twistedmatrix.com/documents/15.5.0/api/twisted.trial.unittest.SynchronousTestCase.html#patch)?